processor/resourcedetectionprocessor/internal/aws/eks/detector.go

```diff
@@ -28,10 +32,15 @@ const (
 	kubernetesServiceHostEnvVar = "KUBERNETES_SERVICE_HOST"
 	authConfigmapNS             = "kube-system"
 	authConfigmapName           = "aws-auth"
+
+	clusterNameAwsEksTag = "aws:eks:cluster-name"
```

Comment:
I would recommend to also check for the cluster name as part of the tag key `kubernetes.io/cluster/CLUSTER_NAME`. I think this tag is most likely to be available on older clusters with self-managed ASGs.

Reply:
Thanks, I'll add.